http/director/proxy.go

```diff
@@ -4,39 +4,67 @@ import (
 	"net/http"
 	"net/http/httputil"
 
+	"fmt"
+
+	"github.com/mwitkow/go-conntrack"
 	"github.com/mwitkow/kedge/http/backendpool"
 	"github.com/mwitkow/kedge/http/director/proxyreq"
 	"github.com/mwitkow/kedge/http/director/router"
 )
 
-func New(pool backendpool.Pool, router router.Router) *Proxy {
+var (
+	AdhocTransport = &(*(http.DefaultTransport.(*http.Transport))) // shallow copy
+)
+
+func New(pool backendpool.Pool, router router.Router, adhoc router.AdhocAddresser) *Proxy {
+	adhocTripper := &(*AdhocTransport) // shallow copy
+	adhocTripper.DialContext = conntrack.NewDialContextFunc(conntrack.DialWithName("adhoc"), conntrack.DialWithTracing())
 	p := &Proxy{
-		reverseProxy: &httputil.ReverseProxy{
+		backendReverseProxy: &httputil.ReverseProxy{
 			Director:  func(r *http.Request) {},
 			Transport: &backendPoolTripper{pool: pool},
 		},
-		router: router,
+		adhocReverseProxy: &httputil.ReverseProxy{
+			Director:  func(r *http.Request) {},
+			Transport: adhocTripper,
+		},
+		router:    router,
+		addresser: adhoc,
 	}
 	return p
 }
 
 type Proxy struct {
-	reverseProxy *httputil.ReverseProxy
-	router       router.Router
+	router    router.Router
+	addresser router.AdhocAddresser
+
+	backendReverseProxy *httputil.ReverseProxy
+	adhocReverseProxy   *httputil.ReverseProxy
 }
 
 func (p *Proxy) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
+	if _, ok := resp.(http.Flusher); !ok {
+		panic("the http.ResponseWriter passed must be an http.Flusher")
```

Comment:
Because this is an internal implementation issue, and you should crash the server if it doesn't work. Basically you passed in a wrong parameter.